src/lib/icon/icon.ts

```diff
@@ -250,4 +281,32 @@ export class MatIcon extends _MatIconMixinBase implements OnChanges, OnInit, Can
   private _cleanupFontValue(value: string) {
     return typeof value === 'string' ? value.trim().split(' ')[0] : value;
   }
+
+  /**
+   * Prepends the current path to all elements that have an attribute pointing to a `FuncIRI`
+   * reference. This is required because WebKit browsers require references to be prefixed with
+   * the current path, if the page has a `base` tag.
+   */
+  private _prependCurrentPathToReferences(element: SVGElement) {
+    // @breaking-change 8.0.0 Remove this null check once `_location` parameter is required.
+    if (!this._location) {
```

Comment:
Last thing: should we just restrict this to webkit, then? I suspect it's not hugely costly, but if we can easily/reliably cut out the extra work, we might as well.

Reply:
I went with doing it on all browsers since it shouldn't be too costly (unless an individual icon has hundreds of elements that all have one of the attributes) and to avoid having too many of these forks in the logic where we sniff out the browser and do special things just for one vendor.

Reply:
Looking a little closer- could we do this prepending at the icon-registry level so that it only has to happen once per svg? I didn't realize earlier it was happening for every directive instance.

Reply:
I left a comment for it a bit further down. Having it in the registry would be ideal, but we want the path at the time the icon was created, rather than when it was fetched. If we did it in the registry, it would be cached with the initial path.

Reply:
I think that technically it can, but the Angular router only reads it once at runtime. Note that we don't add the `base` to the SVG, but we add the current path (`window.location.pathname`). The issue is that if you have a `<base>` tag, you need to prefix the ids with the current path.

Reply:
Doesn't the icon registry not request an SVG until it's used anyway? 

I'm just hesitant to go forward with adding the extra work for pages that might have multiple icons in something like a large data table or option list.

Reply:
It does, but it also caches it afterwards. If performance is a concern, an alternative can be to have some kind of inexpensive check against the string representation of the icon before we do the `querySelector` and we walk through all the attribute. E.g. adding a `MatIconRegistry.hasFuncIRI(iconName)` method which just does a regex match or an `indexOf`.

Reply:
It's not the check so much as actually updating each uri that I'm worried about.

Reply:
I'm not sure whether it would be that expensive, compared to what we're doing for each icon already (cloning the the cached SVG node).

Reply:
That's fair.